flutter/Graminsta/lib/http_service.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:convert';
+import 'package:Graminsta/config.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as baseHttp;
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:Graminsta/constants.dart';
+
+// Class that handles HTTP requests.
+class _HttpService {
+  static _HttpService _instance;
+
+  final globalHeaders = Map<String, String>();
+
+  factory _HttpService() {
+    _instance ??= _HttpService._();
+    return _instance;
+  }
+
+  _HttpService._();
+
+  // Set the auth token for authentication.
+  Future<void> setAuthToken(String token, {Function success}) async {
+    final prefs = await SharedPreferences.getInstance();
+    final setTokenResult = await prefs.setString(userTokenKey, token);
+    if (setTokenResult) {
+      if (success != null) {
+        success();
+      }
+    } else {
+      debugPrint('保存登录token失败');
+    }
+  }
+
+  Future<baseHttp.Response> get(String url,
+      {Map<String, String> headers}) async {
```

Comment:
You can rewrite to something like this.
```
Future<baseHttp.Response> get(
    String url, {
    Map<String, String> headers = const {},
}) async {
    headers.addAll(globalHeaders);
    http.Response response;
    try {
        await http.get(`$baseUrl$url`, headers: headers);
    } catch (e) {
        ...
    }
    return response;
}

Reply:
I think the header default value cannot be set as const{}, or else it will be an unmodifiable Map.

Reply:
No, to provide default value for some container types like `map` and `list` must be const.

> You can also pass lists or maps as default values. The following example defines a function, doStuff(), that specifies a default list for the list parameter and a default map for the gifts parameter.

 See also: https://dart.dev/guides/language/language-tour#optional-parameters